sparkmagic/sparkmagic/livyclientlib/endpoint.py

```diff
@@ -2,19 +2,23 @@
 
 
 class Endpoint(object):
-    def __init__(self, url, username="", password=""):
+    def __init__(self, url, username="", password="", is_default=False):
         if not url:
             raise BadUserDataException(u"URL must not be empty")
         self.url = url.rstrip(u"/")
         self.username = username
         self.password = password
         self.authenticate = False if username == '' and password == '' else True
+        self.is_default = is_default
```

Comment:
Please add comments as to when `is_default` should be true and when it should be false, and implications for it.